### ISSUE TITLE
Neo.ClientError.Security.Forbidden when updating the graph

### ISSUE THREAD

**Issue by HassanHashemi** [neo4j-contrib/neo4j-apoc-procedures#575]
Executing apoc.when(condition, 'statement', '', params) will result into 

`Write operations are not allowed for AUTH_DISABLED with FULL restricted to READ.`

i have 
`dbms.security.procedures.unrestricted=apoc.*`

the query is 

```
with source, destination, tr, m  
      call apoc.when(not exists ((source)-[:TAGS]-()),
              "delete source", "",
              {
              source: source
              }) Yield value
```

**Comment by sarmbruster** [comment]
Use `apoc.do.when` instead if you want to perform write operations. Please give feedback if that solved the issue for you.

**Comment by HassanHashemi** [comment]
@sarmbruster  using apoc.do.when seems to be working.
is it a bug or i misconfigured something?

**Comment by InverseFalcon** [comment]
As an explanation, if a procedure needs to be able to modify the db, it needs to be configured to have write access. However, if we make a single procedure with write access, then users who do not have write access will not be able to call it, even if their usage doesn't actually do writes to the graph.

So we added two versions of the procedure, one that can only execute read operations, and will always be available for users with read-only access (`apoc.when`), and another that can do writes to the graph, and can only be used with users with write capability (`apoc.do.when`). 

There are other APOC Procedures that are similarly split.

Closing.
